test/grouping.jl

```diff
@@ -126,7 +112,7 @@ gd = groupby(df, :g)
 newvar = :n
 
 @testset "Limits of @combine" begin
-    t = @combine(gd, [:i, :g]).i_g_function
+    t = @combine(gd, [:i, :g])[!, 2]
```

Comment:
This means the column name is no longer tested. Same below.

Reply:
Yes, note that this is in the "undefined behavior" tests, so I'm okay with it failing for now. 


The problem is that when DataFrames generates the column name programmatically that relies on the variables `[:i, :g]` being inputted rather than `[:g, :i]`. But since DataFramesMeta uses a `Dict` to store the column names used, this order is not always determined. Hence we can't rely on the function name being the same every time.

Reply:
Got it. Yeah we shouldn't use a dict, or use an ordered dict (in another PR). Couldn't a vector be enough?

Reply:
No, because we attach temporary names like this

```
D[:x] = gensym()
```

and then replace the expression with the temporary names via that same Dict. I don't think the same logic would work without a Dict.

Reply:
Well you could use a vector of pairs and use `findfirst`. For small vectors it could even be faster than a dict.